spec/value_status_spec.rb

```diff
@@ -0,0 +1,112 @@
+require 'spec_helper'
+require 'ronin/recon/value_status'
+require 'ronin/recon/worker'
+
+describe Ronin::Recon::ValueStatus do
+  module TestWorkers
+    class TestWorker < Ronin::Recon::Worker
+    end
+  end
+
+  describe "#initialize" do
+    it "must initialize #values" do
+      expect(subject.values).to eq({})
+    end
+  end
+
+  describe "#value_enqueued" do
+    let(:worker_class)  { TestWorkers::TestWorker }
+    let(:value)         { Ronin::Recon::Values::IP.new('10.12.13.14') }
+    let(:expected_hash) { { foo: :bar, worker_class => :enqueued } }
+
+    context "when #values has value key" do
+      it "must add new key worker_class with value :enqueued" do
+        subject.values[value] = { foo: :bar }
```

Comment:
`#values` is not supposed to be written to from the outside. Instead call `subject.value_enqueued(...)` or `subject.job_started(...)` to simulate status changes for a value. You might also want to add multiple `TestWorker` classes to simulate different workers working on the same value.

```ruby
before do
  subject.value_queued(TestWorkers::Worker1, value)
  subject.value_queued(TestWorkers::Worker2, value)
  ...
end
```